src/Microsoft.ML.LightGbm/LightGbmRankingTrainer.cs

```diff
@@ -153,20 +182,20 @@ private protected override void CheckLabelCompatible(SchemaShape.Column labelCol
         private protected override LightGbmRankingModelParameters CreatePredictor()
         {
             Host.Check(TrainedEnsemble != null, "The predictor cannot be created before training is complete");
-            var innerArgs = LightGbmInterfaceUtils.JoinParameters(Options);
+            var innerArgs = LightGbmInterfaceUtils.JoinParameters(GbmOptions);
             return new LightGbmRankingModelParameters(Host, TrainedEnsemble, FeatureCount, innerArgs);
         }
 
         private protected override void CheckAndUpdateParametersBeforeTraining(IChannel ch, RoleMappedData data, float[] labels, int[] groups)
         {
             Host.AssertValue(ch);
-            Options["objective"] = "lambdarank";
+            GbmOptions["objective"] = "lambdarank";
             ch.CheckValue(groups, nameof(groups));
             // Add default metric.
-            if (!Options.ContainsKey("metric"))
-                Options["metric"] = "ndcg";
+            if (!GbmOptions.ContainsKey("metric"))
```

Comment:
>if (!GbmOptions.ContainsKey("metric")) [](start = 12, length = 38)

we don't have base metrics anymore, so we should provide enum in options and set it value here.